ndc-client/src/models.rs

```diff
@@ -589,19 +619,93 @@ pub struct ExplainResponse {
 
+// ANCHOR: CollectionInsertSchema
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Collection Insert Schema")]
+pub struct CollectionInsertSchema {
+    /// The fields that will be found in the insert row data for the collection and the schema for each field
+    pub fields: BTreeMap<String, InsertFieldSchema>,
+    /// The name of a collection
+    pub collection: String,
+}
+// ANCHOR_END: CollectionInsertSchema
+
+// ANCHOR: InsertFieldSchema
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[schemars(title = "Insert Field Schema")]
+pub enum InsertFieldSchema {
+    ArrayRelation {
+        /// The name of the array relationship over which the related rows must be inserted
+        relationship: String,
+    },
+    Column {
+        /// The name of the column that this field should be inserted into
+        column: String,
+    },
+    ObjectRelation {
+        insertion_order: ObjectRelationInsertionOrder,
+        /// The name of the object relationship over which the related row must be inserted
+        relationship: String,
+    },
+}
+// ANCHOR_END: InsertFieldSchema
+
+// ANCHOR: ObjectRelationInsertionOrder
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize, JsonSchema,
+)]
+#[schemars(title = "Object Relation Insertion Order")]
+#[serde(rename_all = "snake_case")]
+pub enum ObjectRelationInsertionOrder {
+    BeforeParent,
+    AfterParent,
+}
+// ANCHOR_END: ObjectRelationInsertionOrder
+
 // ANCHOR: MutationOperation
 #[skip_serializing_none]
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
 #[schemars(title = "Mutation Operation")]
 #[serde(tag = "type", rename_all = "snake_case")]
 pub enum MutationOperation {
+    Delete {
+        /// The fields to return for the rows affected by this delete operation
+        returning_fields: Option<IndexMap<String, Field>>,
+        /// The name of a collection
+        collection: String,
+        #[serde(rename = "where")]
+        predicate: Option<Expression>,
+    },
+    Insert {
+        post_insert_check: Option<Expression>,
+        /// The fields to return for the rows affected by this insert operation
+        returning_fields: Option<IndexMap<String, Field>>,
+        /// The rows to insert into the collection
+        rows: Vec<BTreeMap<String, serde_json::Value>>,
+        /// The name of a collection
+        collection: String,
+    },
+    Update {
+        post_update_check: Option<Expression>,
+        /// The fields to return for the rows affected by this update operation
+        returning_fields: Option<IndexMap<String, Field>>,
+        /// The name of a collection
+        collection: String,
+        /// The updates to make to the matched rows in the collection
+        updates: Vec<RowUpdate>,
+        #[serde(rename = "where")]
+        r#where: Option<Expression>,
```

Comment:
nit: we should probably call this field `predicate` to match the `Delete` case (assuming these are the same thing?)